common/txmgr/inmemory_store.go

```diff
@@ -0,0 +1,408 @@
+	if err != nil {
+		return nil, fmt.Errorf("new_in_memory_store: %w", err)
+	}
+	for _, fromAddr := range addresses {
+		txs, err := txStore.AllTransactions(ctx, fromAddr, chainID)
+		if err != nil {
+			return nil, fmt.Errorf("address_state: initialization: %w", err)
+		}
+		as, err := NewAddressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE](lggr, chainID, fromAddr, maxUnstarted, txs)
+		if err != nil {
+			return nil, fmt.Errorf("new_in_memory_store: %w", err)
+		}
+
+		ms.addressStates[fromAddr] = as
+	}
+
+	return &ms, nil
+}
+
+// CreateTransaction creates a new transaction for a given txRequest.
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CreateTransaction(
+	ctx context.Context,
+	txRequest txmgrtypes.TxRequest[ADDR, TX_HASH],
+	chainID CHAIN_ID,
+) (
+	txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
+	error,
+) {
+	return txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}, nil
+}
+
+// FindTxWithIdempotencyKey returns a transaction with the given idempotency key
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) FindTxWithIdempotencyKey(ctx context.Context, idempotencyKey string, chainID CHAIN_ID) (*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], error) {
+	return nil, nil
+}
+
+// CheckTxQueueCapacity checks if the queue capacity has been reached for a given address
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CheckTxQueueCapacity(ctx context.Context, fromAddress ADDR, maxQueuedTransactions uint64, chainID CHAIN_ID) error {
+	return nil
+}
+
+// FindLatestSequence returns the latest sequence number for a given address
+// It is used to initialize the in-memory sequence map in the broadcaster
+// TODO(jtw): this is until we have a abstracted Sequencer Component which can be used instead
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) FindLatestSequence(ctx context.Context, fromAddress ADDR, chainID CHAIN_ID) (seq SEQ, err error) {
+	return seq, nil
+}
+
+// CountUnconfirmedTransactions returns the number of unconfirmed transactions for a given address.
+// Unconfirmed transactions are transactions that have been broadcast but not confirmed on-chain.
+// NOTE(jtw): used to calculate total inflight transactions
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CountUnconfirmedTransactions(ctx context.Context, fromAddress ADDR, chainID CHAIN_ID) (uint32, error) {
+	return 0, nil
+}
+
+// CountUnstartedTransactions returns the number of unstarted transactions for a given address.
+// Unstarted transactions are transactions that have not been broadcast yet.
+// NOTE(jtw): used to calculate total inflight transactions
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) CountUnstartedTransactions(ctx context.Context, fromAddress ADDR, chainID CHAIN_ID) (uint32, error) {
```

Comment:
I noticed we have a `CountTransactionsByState` below too. Would we be able to replace both of these with it? We can keep these if there's some custom filtering each of these do that wouldn't make this straight forward